src/libraries/System.Net.Http/tests/FunctionalTests/HttpContentTest.cs

```diff
@@ -509,6 +509,98 @@ public async Task LoadIntoBufferAsync_ThrowIOExceptionInOverriddenAsyncMethod_Th
+                async uri =>
+                {
+                    using HttpClient httpClient = CreateHttpClient();
+
+                    HttpResponseMessage response = await httpClient.GetAsync(
+                        uri,
+                        HttpCompletionOption.ResponseContentRead);
+
+                    var cts = new CancellationTokenSource();
+                    cts.Cancel();
+
+                    await response.Content.LoadIntoBufferAsync(cts.Token);
+                },
+                async server =>
+                {
+                    await server.AcceptConnectionSendResponseAndCloseAsync(content: content);
+                });
+        }
+
+        [Fact]
+        public async Task LoadIntoBufferAsync_Unbuffered_CanBeCanceled_AlreadyCanceledCts()
+        {
+            await LoopbackServer.CreateClientAndServerAsync(
+                async uri =>
+                {
+                    using HttpClient httpClient = CreateHttpClient();
+
+                    HttpResponseMessage response = await httpClient.GetAsync(
+                        uri,
+                        HttpCompletionOption.ResponseHeadersRead);
+
+                    var cts = new CancellationTokenSource();
+                    cts.Cancel();
+
+                    await Assert.ThrowsAsync<TaskCanceledException>(() => response.Content.LoadIntoBufferAsync(cts.Token));
+                },
+                async server =>
+                {
+                    try
+                    {
+                        await server.AcceptConnectionSendResponseAndCloseAsync();
+                    }
+                    catch (Exception ex)
+                    {
+                        _output.WriteLine($"Ignored exception:{Environment.NewLine}{ex}");
+                    }
+                });
+        }
+
+        [Fact]
+        public async Task LoadIntoBufferAsync_Unbuffered_CanBeCanceled()
+        {
+            var cts = new CancellationTokenSource();
+
+            await LoopbackServer.CreateClientAndServerAsync(
+                async uri =>
+                {
+                    using HttpClient httpClient = CreateHttpClient();
+
+                    HttpResponseMessage response = await httpClient.GetAsync(
+                        uri,
+                        HttpCompletionOption.ResponseHeadersRead);
+
+                    await Assert.ThrowsAsync<TaskCanceledException>(() => response.Content.LoadIntoBufferAsync(cts.Token));
+                },
+                async server =>
+                {
+                    await server.AcceptConnectionAsync(async connection =>
+                    {
+                        await connection.ReadRequestHeaderAsync();
+                        await connection.SendResponseAsync(LoopbackServer.GetHttpResponseHeaders(contentLength: 100));
+                        await Task.Delay(250);
+                        cts.Cancel();
+                        await Task.Delay(500);
```

Comment:
Do the delays need to be this long? Might want to mark the test as outerloop.

Reply:
I am not sure but I assume that these delays try to enforce the LoadIntoBuffer task to actively wait on the cancellation token.
For now I have moved it to outerloop, but ideas to make it deterministic are welcome.

Reply:
A deterministic version would look something like a revert of ef6a5a27c4d0cbc1f9b83e6e693740c4c66197ad.
```c#
var observedCancellation = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
```

```c#
var exception = await Assert.ThrowsAsync<TaskCanceledException>(() => task);
observedCancellation.SetResult();
```

```c#
await connection.SendResponseAsync(LoopbackServer.GetHttpResponseHeaders(contentLength: 100));
await Task.Yield();
cts.Cancel();
await observedCancellation.Task.WaitAsync(TestHelper.PassingTestTimeout);
```

But as written, this test isn't enforcing that the client actually honors the cancellation token before receiving the whole response.
The browser implementation seems to be struggling with this (I had to revert a similar change in #104384 due to test failures - [console logs](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-104384-merge-e3831eca814647a093/WasmTestOnFirefox-MT-System.Net.Http.Functional.Tests/1/console.8f17870c.log?helixlogtype=result#:~:text=%5B21%3A14%3A37%5D%20info%3A%20System.AggregateException%20%3A%20One%20or%20more%20errors%20occurred.%20(TypeError%3A%20NetworkError%20when%20attempting%20to%20fetch%20resource.)%20(The%20operation%20has%20timed%20out.))). cc: @pavelsavara 

Given that this test is a copy-paste of existing test logic we have in this file, I'd keep it as-is in this PR.

Reply:
In the browser the server side is played by xharness and driven via WebSocket, that makes it lot slower to react to anything.

It could be probably rewritten in deterministic way, but it could be separate PR if desired.